src/NServiceBus.Persistence.CosmosDB/SynchronizedStorage/SynchronizedStorage.cs

```diff
@@ -28,7 +28,7 @@ protected override void Setup(FeatureConfigurationContext context)
             context.Services.AddSingleton(b => new ContainerHolderResolver(b.GetService<IProvideCosmosClient>(), defaultContainerInformation, databaseName));
 
             context.Services.AddScoped<ICompletableSynchronizedStorageSession, CosmosSynchronizedStorageSession>();
-            context.Services.AddScoped(sp => sp.GetRequiredService<ICompletableSynchronizedStorageSession>().CosmosPersistenceSession());
+            context.Services.AddScoped(sp => (sp.GetService<ISynchronizedStorageSession>() as ICosmosStorageSession)!);
```

Comment:
is there an issue that this doesn't fully cover the code executed by https://github.com/Particular/NServiceBus.Persistence.CosmosDB/blob/8e3dca779506906a2957cedc2348a40328ded587/src/NServiceBus.Persistence.CosmosDB/SynchronizedStorage/SynchronizedStorageSessionExtensions.cs#L21 ?